### DISCUSSION TITLE
need max rectangle coordinates from rectilinear polygon

### DISCUSSION THREAD

**Discussion by Amir4python** [shapely/shapely#1860] [Q&A]
i have sequence of coordinates of a polygon, the polygon has only straight edges no curved edges. we want to find the maximum rectangle that can cover most of the polygon area.

**Comment by ManTouCheer** [comment]
Maybe you can try the last two functions in  https://shapely.readthedocs.io/en/stable/constructive.html.

**Comment by mwtoews** [comment]
If you define a "max" rectangle that is aligned to the coordinate system, this can be built with the `.bounds` attribute:
```python
import shapely
from shapely.geometry import box

geom = shapely.from_wkt("POLYGON ((0 0, -50 86.6025, 123.20508 186.6025, 173.205 100, 0 0))")

box(*geom.bounds)
# <POLYGON ((173.205 0, 173.205 186.602, -50 186.602, -50 0, 173.205 0))>
```
blue is input `geom` and red is the max bounds rectangle:
![image](https://github.com/shapely/shapely/assets/895458/0061f9b1-ca22-4d16-9608-8b27dfc64fc6)

or re-evaluate the minimum rotated rectangle
```python
shapely.minimum_rotated_rectangle(geom)
# <POLYGON ((-50 86.602, 0 0, 173.205 100, 123.205 186.603, -50 86.602))>
```
which is _almost_ the same geometry (there are minor differences due the algorithm), since the input was already rectangular.

**Reply by jorisvandenbossche** [reply]
@mwtoews btw for getting the bounding box as a geometry (and not 4 numbers), you can also directly use `envelope()` method/function, instead of passing the bounds to `box`

**Comment by Amir4python** [comment]
[image: image.png]

the white space is the the polygon, from that white polygon, i need to max
area covered rectangle that covers most of the white space., in Landscape
or portrait or Both.

**Reply by jorisvandenbossche** [reply]
@Amir4python the image is not showing for me

**Reply by Amir4python** [reply]
![Screenshot 2023-08-01 175758](https://github.com/shapely/shapely/assets/60217471/8451e938-8e1d-4e31-b8da-f66630725359)

i need to find maximum rectangle that can eat up yellow region.
for Example 
![Screenshot 2023-08-03 110241](https://github.com/shapely/shapely/assets/60217471/c2ff4f35-2762-4a63-be67-50b93251594e)

 
here i can blue rectangle that eats up most of yellow area.

**Reply by Amir4python** [reply]
@jorisvandenbossche is the image visible.
